src/app/repository/cpe-api.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+
+interface iCpeApiService {
+  getProject(id: string): Promise<any>;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CpeApiService implements iCpeApiService {
+  constructor() { }
+
+  /**
+   * gets a list of projects from the api route
+   */
+  async getProjects() {
+    return fetch('http://localhost:3000/projects');
+  }
+
+  /**
+   * gets a projects data via the api route
+   */
+  async getProject(id: string) {
+    return fetch(`http://localhost:3000/projects/get/${id}`);
+  }
+
+  /**
+   * gets a pages data via the api route
+   */
+  async getPage(id: string) {
+    return fetch(`http://localhost:3000/pages/get/${id}`);
+  }
```

Comment:
see above

Reply:
Yeah, indeed it doesen't meed to be async. I'll change it. But we want to use https://github.com/Effect-TS/effect  anyways - so no fetch or htttpclient at all. Was not of any concern to me.